config.go

```diff
@@ -5,28 +5,44 @@ import (
 )
 
 const (
-	apiURLv1                       = "https://api.openai.com/v1"
+	openaiAPIURLv1                 = "https://api.openai.com/v1"
 	defaultEmptyMessagesLimit uint = 300
+
+	azureAPIPrefix         = "openai"
+	azureDeploymentsPrefix = "deployments"
+)
+
+type APIType string
+
+const (
+	APITypeOpenAI  APIType = "OPEN_AI"
+	APITypeAzure   APIType = "AZURE"
+	APITypeAzureAD APIType = "AZURE_AD"
 )
 
 // ClientConfig is a configuration of a client.
 type ClientConfig struct {
 	authToken string
+	BaseURL   string
+	OrgID     string
 
-	HTTPClient *http.Client
+	APIType    APIType
+	APIVersion string // required for APITypeAzure or APITypeAzureAD
+	Engine     string // required for APITypeAzure or APITypeAzureAD
 
-	BaseURL string
-	OrgID   string
+	HTTPClient *http.Client
 
 	EmptyMessagesLimit uint
 }
 
 func DefaultConfig(authToken string) ClientConfig {
```

Comment:
Do we want to have a `DefaultAzure` config (non-AD) so it would fill the config with the necessary fields?

Reply:
I can add that.